src/librustdoc/html/render/mod.rs

```diff
@@ -504,23 +523,36 @@ impl<'tcx> FormatRenderer<'tcx> for Context<'tcx> {
             current: Vec::new(),
             dst,
             render_redirect_pages: false,
-            id_map: Rc::new(RefCell::new(id_map)),
-            deref_id_map: Rc::new(RefCell::new(FxHashMap::default())),
-            shared: Arc::new(scx),
-            all: Rc::new(RefCell::new(AllTypes::new())),
-            errors: Rc::new(receiver),
+            id_map: RefCell::new(id_map),
```

Comment:
This is passed in as a parameter to `init()`, but it was and is useless because it is cleared before rendering each item. It originally comes from the handler for `external_html`. I feel a bit confused about it because only one `Context` is ever initialized explicitly—all the others are derived from it—and each one is local to a page. So passing in the `external_html` `id_map` doesn't really work. Should I just remove that parameter so that it's never passed in in the first place or should I leave it in case it makes fixing the bug easier?

Note that there don't seem to be any tests for `external_html` :/

Reply:
cc https://github.com/rust-lang/rust/pull/82356#discussion_r585206029

I think Joshua said I can do that in a later PR actually: https://github.com/rust-lang/rust/pull/82356#discussion_r585208715